main.go

```diff
@@ -265,20 +280,31 @@ tagLoop:
 		if len(img.DockerfileExtras) == 0 {
 			source := fmt.Sprintf("%s%s:%s", dockerTransport, img.Image, tag)
 			wg := sync.WaitGroup{}
-			wg.Add(3)
+			wg.Add(4)
 			// Quay
 			destination := fmt.Sprintf("%s%s/%s:%s", dockerTransport, quayURL, destinationName, destinationTag)
 			go copyImage(&wg, errorCounter, source, destination)
 			// Azure
 			destination = fmt.Sprintf("%s%s/%s:%s", dockerTransport, azureURL, destinationName, destinationTag)
 			go copyImage(&wg, errorCounter, source, destination)
+			// Azure upstream
+			destination = fmt.Sprintf("%s%s/%s:%s", dockerTransport, azureUpstreamURL, upstreamName, tag)
+			go copyImage(&wg, errorCounter, source, destination)
 			// Aliyun
 			destination = fmt.Sprintf("%s%s/%s:%s", dockerTransport, aliyunURL, destinationName, destinationTag)
 			go copyImage(&wg, errorCounter, source, destination)
 			wg.Wait()
 
 			// We'll skip to the next tag
 			continue
+		} else {
+			source := fmt.Sprintf("%s%s:%s", dockerTransport, img.Image, tag)
+			wg := sync.WaitGroup{}
+			wg.Add(1)
+			// AzureCR/upstream
+			destination := fmt.Sprintf("%s%s/%s:%s", dockerTransport, azureUpstreamURL, upstreamName, tag)
```

Comment:
Same, shouldn't this use `destinationTag`? Or if we want to keep the original upstream tag without potential suffix changes, shouldn't we assign both to indicate that these are the same images?